WarOfEmpires.CommandHandlers.Tests/Empires/HealTroopsCommandHandlerTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NSubstitute;
+using WarOfEmpires.CommandHandlers.Empires;
+using WarOfEmpires.Commands.Empires;
+using WarOfEmpires.Domain.Common;
+using WarOfEmpires.Domain.Players;
+using WarOfEmpires.Domain.Security;
+using WarOfEmpires.Repositories.Players;
+using WarOfEmpires.Test.Utilities;
+
+namespace WarOfEmpires.CommandHandlers.Tests.Empires {
+
+    [TestClass]
+    public sealed class HealTroopsCommandHandlerTests {
+        private readonly FakeWarContext _context = new FakeWarContext();
+        private readonly PlayerRepository _repository;
+        private readonly Player _player;
+
+        public HealTroopsCommandHandlerTests() {
+            _repository = new PlayerRepository(_context);
+
+            var user = Substitute.For<User>();
+            user.Email.Returns("test@test.com");
+            user.Status.Returns(UserStatus.Active);
+
+            var player = Substitute.For<Player>();
+            player.Stamina.Returns(90);
+            player.User.Returns(user);
+            player.CanAfford(Arg.Any<Resources>()).Returns(true);
+            player.Archers.Returns(new Troops(10,2));
+            player.Cavalry.Returns(new Troops(0,0));
+            player.Footmen.Returns(new Troops(0,0));
+
+            _context.Users.Add(user);
+            _context.Players.Add(player);
+            _player = player;
+        }
+
+        [TestMethod]
+        public void HealTroopsCommandHandler_Succeeds() {
+            var handler = new HealTroopsCommandHandler(_repository);
+            var command = new HealTroopsCommand("test@test.com", "5");
+
+            var result = handler.Execute(command);
+
+            result.Success.Should().BeTrue();
+            _player.Received().HealTroops(5);
+            _context.CallsToSaveChanges.Should().Be(1);
+        }
+
+        [TestMethod]
+        public void HealTroopsCommandHandler_AllowsEmptyStamina() {
```

Comment:
See comment on the command; if you agree this test can go (or change)

Reply:
Test removed after updating the regular expression and confirming that it catches an empty string. 
Have also renamed the other tests in HealTroopsCommandHandlerTests with underscores to conform to the naming convention